redis_queue_engine.go

```diff
@@ -0,0 +1,148 @@
+package ewf
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"sync"
+	"time"
+
+	"github.com/redis/go-redis/v9"
+)
+
+var _ QueueEngine = (*RedisQueueEngine)(nil)
+
+// RedisQueueEngine is the Redis implementation of the QueueEngine interface
+type RedisQueueEngine struct {
+	client *redis.Client
+	mu     sync.Mutex
+	queues map[string]*RedisQueue
+}
+
+// NewRedisQueueEngine creates a new RedisQueueEngine with the given Redis address
+func NewRedisQueueEngine(address string) *RedisQueueEngine {
+	client := redis.NewClient(&redis.Options{
+		Addr: address,
+	})
```

Comment:
`NewRedisQueueEngine` should just receive a redis client instead of the address. User can handle how the client is configured (user, pass, pool, etc..) and the queue engine just operates on it directly